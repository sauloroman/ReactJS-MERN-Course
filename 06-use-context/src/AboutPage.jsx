import { useContext } from "react"
import { UserContext } from "./context/UserContext"

export const AboutPage = () => {

  const { user } = useContext(UserContext)

  return (
    <div>
      <h1 className="heading__page">About Page</h1>

      <div className="user">
        <p className="user__name">{user.name}</p>
        <p className="user__email">{user.email}</p>
        <p className="user__age">{user.age}</p>
      </div>
    </div>
  )
}
