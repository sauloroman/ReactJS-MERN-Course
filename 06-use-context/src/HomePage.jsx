export const HomePage = () => {
  return (
    <div>
      <h1 className="heading__page">Home Page</h1>
    </div>
  )
}
