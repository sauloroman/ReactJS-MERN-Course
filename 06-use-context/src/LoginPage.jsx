import { useContext } from "react"
import { UserContext } from "./context/UserContext"

export const LoginPage = () => {

  const { setUser } = useContext(UserContext)

  return (
    <div>
      <h1 className="heading__page">Bienvenido a la app</h1>
      <button 
        className="btn"
        onClick={() => setUser({ name: 'Saulo Román', email: 'romansantillan1998@outlook.com', age: 26, })}>Iniciar Sesión</button>
    </div>
  )
}
