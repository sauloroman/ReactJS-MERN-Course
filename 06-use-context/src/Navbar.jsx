import { NavLink } from 'react-router-dom'

export const Navbar = () => {
  return (
    <div className="nav">
      <ul className="nav__list">
        <NavLink to={'/'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>Home</li>
        </NavLink>
        <NavLink to={'/about'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>About</li>
        </NavLink>
        <NavLink to={'/login'} className={ ({isActive}) => `${isActive && 'nav__item--active'} nav__item` }>
          <li className='nav__text'>Login</li>
        </NavLink>
      </ul>
    </div>
  )
}
