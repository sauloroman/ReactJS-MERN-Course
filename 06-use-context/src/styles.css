*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%;
}

body {
  box-sizing: border-box;
  font-size: 1.6rem;
  font-family: 'Poppins', sans-serif;
  color: #555;
}

.container {
  margin: 0 auto;
  width: 120rem;
  max-width: 95%;
}

.heading {
  font-size: 3.2rem;
  color: #5959ec;
  margin-bottom: 2rem;
}

.heading__page {
  font-size: 2rem;
  color: #000;
}

.nav {
  background-color: #000;
  margin-bottom: 2rem;
}

.nav__list {
  list-style: none;
  display: flex;
  justify-content: space-around;
}

.nav__item {
  width: 100%;
  text-align: center;
  padding: 1rem 0;
  color: #fff;
  text-decoration: none;
}

.btn {
  background-color: #5959ec;
  font-family: inherit;
  color: #fff;
  padding: 1rem 2rem;
  border: none;
  cursor: pointer;
}