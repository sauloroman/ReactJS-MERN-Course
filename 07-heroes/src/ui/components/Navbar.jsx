import { NavLink, Link, useNavigate } from 'react-router-dom'

export const Navbar = () => {

  const navigate = useNavigate()

  const onLogout = () => {
    navigate('/login', { replace: true })
  }

  return (
    <nav className="nav">
      <ul className="nav__list">
        <Link className='nav__link' to={'/marvel'}>Asociaciones</Link>
        <NavLink to={'/marvel'} className={({isActive}) => `nav__link ${isActive && 'nav__link--active'}`}>
          <li>Marvel</li>
        </NavLink>
        <NavLink to={'/dc'} className={({isActive}) => `nav__link ${isActive && 'nav__link--active'}`}>
          <li>DC</li>
        </NavLink>
        <NavLink to={'/search'} className={({isActive}) => `nav__link ${isActive && 'nav__link--active'}`}>
          <li>Buscar</li>
        </NavLink>
      </ul>

      <div className="nav__info">
        <div className="nav__name">Rom√°n</div>
        <button onClick={ onLogout } className="nav__button">Salir</button>
      </div>
    </nav>
  )
}
