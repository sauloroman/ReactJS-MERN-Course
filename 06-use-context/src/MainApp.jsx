import { Navigate, Route, Routes } from "react-router-dom"
import { Navbar } from "./Navbar"
import { HomePage } from "./HomePage"
import { AboutPage } from "./AboutPage"
import { LoginPage } from "./LoginPage"
import { UserProvider } from "./context/UserProvider"

export const MainApp = () => {
  return (
    <UserProvider>
      <div className="container">
        <h1 className="heading">UseContext</h1>
        <Navbar />

        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/about" element={<AboutPage />} />
          <Route path="/login" element={<LoginPage />} />
          <Route path="/*" element={<Navigate to={'/'}/>} />
        </Routes>
      </div>
    </UserProvider>
  )
}
