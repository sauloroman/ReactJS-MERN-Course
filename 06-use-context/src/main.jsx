import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { MainApp } from './MainApp'
import { BrowserRouter } from 'react-router-dom'
import './styles.css'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <BrowserRouter>
      <MainApp />
    </BrowserRouter>
  </StrictMode>,
)
