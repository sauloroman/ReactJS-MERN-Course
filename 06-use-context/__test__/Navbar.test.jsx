import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Navbar } from '../src/Navbar'

describe('Tests in Navbar Component', () => {

  test('it must match with snapshot', () => {

    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )

    expect( container ).toMatchSnapshot()

  })

  test('it should have the options to navigate to Home, About and Login page', () => {

    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )

    const homeOption = screen.getByText('Home')
    const aboutOption = screen.getByText('About')
    const loginOption = screen.getByText('Login')

    expect( homeOption ).toBeTruthy()
    expect( aboutOption ).toBeTruthy()
    expect( loginOption ).toBeTruthy()

  })

})