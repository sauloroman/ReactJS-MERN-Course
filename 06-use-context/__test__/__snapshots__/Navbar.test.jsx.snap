// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Navbar Component it must match with snapshot 1`] = `
<div>
  <div
    class="nav"
  >
    <ul
      class="nav__list"
    >
      <a
        aria-current="page"
        class="nav__item--active nav__item"
        href="/"
      >
        <li
          class="nav__text"
        >
          Home
        </li>
      </a>
      <a
        class="false nav__item"
        href="/about"
      >
        <li
          class="nav__text"
        >
          About
        </li>
      </a>
      <a
        class="false nav__item"
        href="/login"
      >
        <li
          class="nav__text"
        >
          Login
        </li>
      </a>
    </ul>
  </div>
</div>
`;
